server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -192,6 +190,94 @@ private Mapping toMapping(Username rewrittenUsername, Type type) {
         throw new IllegalArgumentException("unhandled enum type");
     }
 
+    @Override
+    public List<MappingSource> getResolvedMappingSources(String user, Domain domain, EnumSet<Type> mappingTypes) throws ErrorMappingException, RecipientRewriteTableException {
+        return getMappingSources(Username.fromLocalPartWithDomain(user, domain), mappingLimit, mappingTypes);
+    }
+
+    private List<MappingSource> getMappingSources(Username username, int mappingLimit, EnumSet<Type> mappingTypes) throws TooManyMappingException, RecipientRewriteTableException {
```

Comment:
I believe the `int mappingLimit` parameter can safely be removed as it always have the same value because this method is only called once.